src/java.base/share/classes/java/util/Arrays.java

```diff
@@ -77,6 +79,27 @@ public final class Arrays {
     // Suppresses default constructor, ensuring non-instantiability.
     private Arrays() {}
 
+    /**
+     * Sorts the specified array into ascending numerical order.
+     *
+     *
+     * @param elemType the class of the array to be sorted
+     * @param array the array to be sorted
+     * @param offset the relative offset, in bytes, from the base address of
+     * the array to sort, otherwise if the array is {@code null},an absolute
+     * address pointing to the first element to sort from.
+     * @param fromIndex the index of the first element, inclusive, to be sorted
+     * @param toIndex the index of the last element, exclusive, to be sorted
+     */
+    @IntrinsicCandidate
+    public static void arraySort(Class<?> elemType, Object array, long offset, int fromIndex, int toIndex) {
+        if (elemType == int.class) DualPivotQuicksort.sort((int[]) array, 0, fromIndex, toIndex);
+        else if (elemType == long.class) DualPivotQuicksort.sort((long[]) array, 0, fromIndex, toIndex);
+        else if (elemType == float.class) DualPivotQuicksort.sort((float[]) array, 0, fromIndex, toIndex);
+        else if (elemType == double.class) DualPivotQuicksort.sort((double[]) array, 0, fromIndex, toIndex);
+        else throw new UnsupportedOperationException("arraySort intrinsic not supported for this type: " + elemType.toString());
```

Comment:
I'm curious if there is a performance difference using switch pattern on element type that would generate an `invokedynamic typeSwitch` over the primitive array types e.g.:

```suggestion
        switch (array) {
            case int[] arr -> DualPivotQuicksort.sort(arr, 0, fromIndex, toIndex);
            case long[] arr -> DualPivotQuicksort.sort(arr, 0, fromIndex, toIndex);
            case float[] arr -> DualPivotQuicksort.sort(arr, 0, fromIndex, toIndex);
            case double[] arr -> DualPivotQuicksort.sort(arr, 0, fromIndex, toIndex);
            default -> throw new UnsupportedOperationException(
                    "arraySort intrinsic not supported for this type: " + elemType);
        }
```

Reply:
What is the reasoning behind this new public API? It doesn't follow the usual Java convention, which is to have overloads for each type. And it doesn't seem to provide anything not already provided by `Arrays.sort()`.

Reply:
Hi Andrew, the reason for the public API is to make AVX512 sort available to other data structures like MemorySegment (including the ones backed by native heap). The API of the arraySort() AVX512 intrinsic is similar to the public API of ArraysSupport.vectorizedMismatch() which is used by MemorySegment.mismatch().

Reply:
There's no need to make this method public, and it should not be. There is no need to have it in the Java API. `ArraysSupport.vectorizedMismatch()` is in an internal JDK class, not part of the Java API.

Reply:
Sure Andrew. Will make this method private as suggested.